service/worker/scanner/scanner.go

```diff
@@ -102,8 +103,9 @@ type (
 	// of database tables to cleanup resources, monitor anamolies
 	// and emit stats for analytics
 	Scanner struct {
-		context scannerContext
-		wg      sync.WaitGroup
+		context      scannerContext
+		wg           sync.WaitGroup
+		shutdownOnce channel.ShutdownOnce
```

Comment:
How is this different/better than storing a Context and a CancelFunc in the Scanner and cancelling it on Stop? (and using that Context as the base context for ThrottleRetryContext, WithTimeout, etc.) It seems like that's a little less code and doesn't require familiarity with this other library

Reply:
Also, encapsulating this behavior in ShutdownOnce makes this reusable